From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rob Clark <rob.clark@oss.qualcomm.com>
Date: Fri, 23 Oct 2026 10:22:18 -0700
Subject: [PATCH] freedreno/a6xx: Fix fbfetch with msrtt

We need to use the # of samples in GMEM, not the # of samples of the
underlying resource.

Cc: mesa-stable
Closes: https://gitlab.freedesktop.org/mesa/mesa/-/work_items/16656
Signed-off-by: Rob Clark <rob.clark@oss.qualcomm.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/45564>
(cherry picked from commit cf8bd35eae3928ad88c5b29e5921cab6281b9dad)
---
 src/gallium/drivers/freedreno/a6xx/fd6_gmem.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gallium/drivers/freedreno/a6xx/fd6_gmem.cc b/src/gallium/drivers/freedreno/a6xx/fd6_gmem.cc
index d725351325e0..2115a5f4ee12 100644
--- a/src/gallium/drivers/freedreno/a6xx/fd6_gmem.cc
+++ b/src/gallium/drivers/freedreno/a6xx/fd6_gmem.cc
@@ -411,7 +411,7 @@ patch_fb_read_gmem(struct fd_batch *batch)
          uint32_t descriptor[FDL6_TEX_CONST_DWORDS] = {
                A6XX_TEX_MEMOBJ_0_FMT(fd6_texture_format(
                      format, (enum a6xx_tile_mode)rsc->layout.tile_mode, false)) |
-               A6XX_TEX_MEMOBJ_0_SAMPLES(fd_msaa_samples(prsc->nr_samples)) |
+               A6XX_TEX_MEMOBJ_0_SAMPLES(fd_msaa_samples(pfb->samples)) |
                A6XX_TEX_MEMOBJ_0_SWAP(WZYX) |
                A6XX_TEX_MEMOBJ_0_TILE_MODE(TILE6_2) |
                COND(util_format_is_srgb(format), A6XX_TEX_MEMOBJ_0_SRGB) |
@@ -446,7 +446,7 @@ patch_fb_read_gmem(struct fd_batch *batch)
 
             A8XX_TEX_MEMOBJ_2_WIDTH(pfb->width) |
             A8XX_TEX_MEMOBJ_2_HEIGHT(pfb->height) |
-            A8XX_TEX_MEMOBJ_2_SAMPLES(fd_msaa_samples(prsc->nr_samples)),
+            A8XX_TEX_MEMOBJ_2_SAMPLES(fd_msaa_samples(pfb->samples)),
 
             A8XX_TEX_MEMOBJ_3_FMT(fd6_texture_format(
                      format, (enum a6xx_tile_mode)rsc->layout.tile_mode, false)) |
